File: backend/app/schemas/risk_assessment.py
from datetime import datetime
from typing import Any
from pydantic import BaseModel
from app.schemas.common import DatetimeStr

class RiskAssessmentGenerateRequest(BaseModel):
    custom_instruction: str | None = None

class RiskAssessmentSummary(BaseModel):
    risk_source_count: int = 0
    risk_level_distribution: dict[str, int] = {}
    top_risks: list[dict] = []
    risk_by_category: dict[str, int] = {}
    key_findings: list[str] = []
    overall_assessment: str = ""

class RiskAssessmentReportResponse(BaseModel):
    id: str
    enterprise_id: str
    title: str
    content: str
    # JSONB summary 结构不定（chapters/images/统计键等），需原样透传，
    # 否则草稿章节/四色图等扩展键会被 Pydantic 白名单过滤掉。
    summary: dict[str, Any]
    status: str
    generated_by: str
    generated_at: DatetimeStr | None
    created_at: DatetimeStr
    updated_at: DatetimeStr
    model_config = {"from_attributes": True}

class RiskAssessmentPreviewResponse(BaseModel):
    report_id: str
    title: str
    html: str

File: frontend/src/types/riskAssessment.ts
export interface RiskAssessmentSummary {
  risk_source_count: number;
  risk_level_distribution: Record<string, number>;
  top_risks: Array<{
    name: string;
    category: string;
    risk_level: string;
    likelihood: string;
    severity: string;
    location: string;
    key_control_measures: string;
  }>;
  risk_by_category: Record<string, number>;
  key_findings: string[];
  overall_assessment: string;
}

export interface RiskAssessmentReport {
  id: string;
  enterprise_id: string;
  title: string;
  content: string;
  summary: RiskAssessmentSummary;
  status: "draft" | "generating" | "completed";
  generated_by: "ai" | "manual";
  generated_at: string | null;
  current_version?: number;
  created_at: string;
  updated_at: string;
}

export interface RiskAssessmentPreview {
  report_id: string;
  title: string;
  html: string;
}

export interface ReportVersionItem {
  id: string;
  version_number: number;
  created_by: string;
  description: string | null;
  created_at: string;
}

export interface SSEEvent {
  type: "progress" | "chunk" | "section_done" | "batch_done" | "error" | "token" | "chapter_start" | "chapter_end" | "done" | "complete";
  message?: string;
  stage?: string;
  content?: string;
  section_key?: string;
  report_id?: string;
  title?: string;
  current?: number;
  total?: number;
  completed?: number;
  failed?: number;
  chapter?: string;
  token?: string;
  chunk?: string;
  chapters?: Array<{ key: string; title: string; content: string }>;
}
